src/components/FriendRequests.tsx

```diff
@@ -0,0 +1,70 @@
+'use client'
+
+import {FC, useState} from "react";
+import {Check, UserPlus, X} from 'lucide-react';
+import axios from "axios";
+import { useRouter } from 'next/navigation';
+
+interface FriendRequestsProps {
+    incomingFriendRequests: {
+        senderId:string,
+        senderEmail:string
+    }[]
+}
+
+type command = 'accept' | 'deny'
+
+const FriendRequests :FC<FriendRequestsProps> =({incomingFriendRequests})=>{
+    const router = useRouter()
+    const [requests, setRequests] = useState<{
+        senderId:string,
+        senderEmail:string
+    }[]>(incomingFriendRequests);
+
+    const apiPost = async (senderId: string, cmd: command) =>{
+        await axios.post(`/api/friends/${cmd}`, {id: senderId});
+        setRequests(previousOptions => previousOptions.filter(item=> item.senderId != senderId));
+        router.refresh();
+    }
+
+    const accept = async (senderId: string)=>{
+        await apiPost(senderId, 'accept')
+    }
+
+    const deny = async (senderId: string)=>{
+        await apiPost(senderId, 'deny')
+    }
+
+    return <div aria-label='friend requests'>
+        {incomingFriendRequests.length == 0 ?
+        <p className='friend-requests-nothing'>
+            Nothing to show here...
+        </p>:
+            requests.map((r)=>{
+                return (<div className='friend-requests' key={r.senderId}>
+                    <UserPlus  aria-label='add user'/>
```

Comment:
If want to switch to lucide-react icons, need to implement the change in Icons.tsx too.
Either pull a new svg from the library or fix Icons so it uses UserPlus there